### PR TITLE
feat(cross-sdk-tests): pin v0.3.0 vectors for envelope + peer_credential + emitter_metadata

### PR DESCRIPTION
**Pull request by ojongerius**
## What

Pins cross-SDK test vectors for the v0.3.0 protocol shape introduced by spec PR #496:

- `cross-sdk-tests/v030_vectors.json` — two signed receipts:
  - `parametersDisclosureEnvelopeReceipt` — `action.parameters_disclosure` carries the HPKE asymmetric envelope (ADR-0012 amendment). Envelope bytes are taken from `spec/test-vectors/disclosure-envelope/vectors.json` vector-1, which the Go SDK already pins byte-for-byte against RFC 9180 §A.1.1 / RFC 7748 §6.1 Alice in `sdk/go/receipt/disclosure_test.go:TestDeterministicVector1`.
  - `peerCredentialEmitterMetadataReceipt` — exercises the new daemon-attested fields: `action.peer_credential` (linux POSIX: pid 12345, uid/gid 1000, exe_path) plus `action.emitter_metadata.drop_count: 3` (synthetic events_dropped style per ADR-0010).
- `cross-sdk-tests/v030_vectors_test.go` — `-tags=integration` runner that schema-validates each receipt against `spec/schema/agent-receipt.schema.json`, verifies the Ed25519 signature against the shared test key, and reproduces the pinned `expectedReceiptHash` byte-for-byte via the Go SDK's `Canonicalize` + SHA-256.
- `cross-sdk-tests/cmd/generate-vectors/main.go` — extended with a `generateV030Vectors` step that builds the receipts as `map[string]any` trees so the new envelope and daemon-attested shapes do not require touching `sdk/go/receipt/types.go` (PR-C's scope). Signs with low-level `crypto/ed25519` mirroring the SDK's PEM-parse + sign-the-JCS-bytes flow.
- `cross-sdk-tests/README.md` — documents `v020_vectors.json` and the new `v030_vectors.json` alongside the existing TS/Go/Py vector files.

Frozen by design: `v020_vectors.json` is untouched. The v0.3.0 schema's `oneOf` on `parameters_disclosure` continues to accept the legacy flat-map shape, so v020 receipts still schema-validate against the post-#496 schema and their proofs do not need regenerating.

## Why

PR-B of the parameters_disclosure Phase 2 migration tracked under #280. Spec PR-A (#496) just landed; PR-C (Go SDK + daemon), PR-D (TS SDK), and PR-E (Python SDK) all depend on having these pinned vectors so each SDK can prove byte-identical reproduction of the new envelope and daemon-attested shapes before they ship the typed code paths.

Refs #280, #496, #468 (Go HPKE), #472 (TS HPKE), #494 (Python HPKE).

## Checklist

- [x] Tests pass for all changed components — `go test -tags=integration ./...` is green for both `cross-sdk-tests/` (incl. 4 new sub-cases) and `sdk/go/...`.
- [x] Linter passes (`go vet ./...` clean; `gofmt` no longer reports drift — opportunistic whitespace cleanup of `cmd/generate-malformed-vectors/main.go` struct tags is included as a papercut fix).
- [x] No real keys or secrets in the diff — Ed25519 test keypair is the same well-known fixture already used by all v020 vectors; HPKE envelope reuses spec vector-1 (RFC 7748 §6.1 Alice public key).
- [x] Cross-language tests pass (the new file is meant to be wired into TS / Py by PR-D / PR-E; the Go side is wired here and green).
- [ ] AGENTS.md updated — N/A (no project-structure change).
- [ ] Spec changes have been reviewed by a maintainer — N/A (PR-B touches no spec files; rides on the already-accepted PR #496).

## Security

- [x] This PR touches crypto, auth, or secrets handling — receipt signatures and HPKE envelope bytes.
- [x] Primitives and parameters have been reviewed — Ed25519 signing reuses the SDK's existing path (PEM PKCS8 → `ed25519.Sign` over RFC 8785 JCS bytes); HPKE envelope is the byte-pinned spec vector-1, not a freshly encrypted value.
- [x] All inputs are validated at trust boundaries — the new test asserts proof type, multibase prefix, and signature length before verifying.
- [x] Edge cases are tested — both receipt shapes go through the same schema/signature/hash pipeline as the existing v020 vectors.

### CODE REVIEW

File: cross-sdk-tests/cmd/generate-vectors/main.go [hunk 1]
```diff
@@ -1,12 +1,17 @@
 // generate_go_vectors reads ts_vectors.json, signs the same unsigned receipt
-// with the Go SDK using the shared keypair, and writes go_vectors.json and
-// v020_vectors.json.
+// with the Go SDK using the shared keypair, and writes go_vectors.json,
+// v020_vectors.json, and v030_vectors.json.
```
[the remark this answers is not in the document]
Comment (ojongerius) [reply cross-sdk-tests/cmd/generate-vectors/main.go]: Fixed in a0079c4 — rewrote the file header to enumerate each output and call out that only `go_vectors.json` is derived from `ts_vectors.json`; `v020_vectors.json` and `v030_vectors.json` are independently constructed and reuse only the shared keypair.
